src/main/java/picard/illumina/parser/TileMetricsUtil.java

```diff
@@ -123,25 +123,23 @@ public static File renderTileMetricsFileFromBasecallingDirectory(final File illu
                 // If both the phasing and pre-phasing data are missing, then likely something went wrong when imaging
                 // this tile, for example a grain of sand disrupting the path of light to the sensor.  If only one of them
                 // is missing, then likely the data is corrupt.
-                if (!codeMetricsMap.containsKey(phasingCode) && !codeMetricsMap.containsKey(prePhasingCode)) {
+                if (codeMetricsMap.containsKey(phasingCode) && codeMetricsMap.containsKey(prePhasingCode)) {
+                    phasingValue = CollectionUtil.getSoleElement(codeMetricsMap.get(phasingCode)).getMetricValue();
```

Comment:
Could you make `phasingValue` and `prePhasingValue` final and assign to zero in the `else` block?